src/components/LibraryStats.tsx

```diff
@@ -1,180 +1,294 @@
-                    barSize={50}
-                    fill="#000000"
-                  />
-                </BarChart>
-              </li>
-            )}
-          </ul>
-        )}
-      </div>
-    );
-  }
-
-  formatNumber(n) {
-    return n ? numeral(n).format("0.[0]a") : 0;
-  }
-}
+ for a single library or all libraries the admin has access to. */
+const LibraryStats = (props: LibraryStatsProps) => {
+  const {
+    stats: {
+      name: libraryName,
+      key: libraryKey,
+      collections,
+      inventorySummary: inventory,
+      patronStatistics: patrons,
+    },
+    library,
+  } = props;
+
+  const chartItems =
+    props.stats &&
+    props.stats.collections &&
+    collections
+      .map(({ name, inventory }) => ({ name, ...inventory }))
+      .sort((a, b) => (a.name.toLowerCase() > b.name.toLowerCase() ? 1 : -1));
+
+  /* Customize the Rechart tooltip to provide additional information */
+  const CustomTooltip = (props) => {
+    const { active, payload } = props;
+    if (!active) return null;
+
+    // Nab inventory data from one of the chart payload objects.
+    const chartInventory = payload[0].payload;
+    const aboveTheLineColor = "#030303";
+    const belowTheLineColor = "#A0A0A0";
+    const aboveTheLine: chartTooltipData[] = [
+      {
+        dataKey: "titles",
+        name: inventoryKeyToLabelMap.titles,
+        value: chartInventory.titles,
+      },
+      {
+        dataKey: "availableTitles",
+        name: inventoryKeyToLabelMap.availableTitles,
+        value: chartInventory.availableTitles,
+      },
+      ...payload.filter(({ value }) => value > 0),
+    ].map((entry) => ({ ...entry, color: aboveTheLineColor }));
+    const aboveTheLineKeys = [
+      "name",
+      ...aboveTheLine.map(({ dataKey }) => dataKey),
+    ];
+    const belowTheLine = Object.entries(chartInventory)
+      .filter(([key]) => !aboveTheLineKeys.includes(key))
+      .map(([key, value]) => ({
+        dataKey: key,
+        name: inventoryKeyToLabelMap[key],
+        value,
+        color: belowTheLineColor,
+      }));
+    const newPayload = [
+      ...aboveTheLine,
+      {}, // blank line
+      { value: ">>> Additional Information <<<", color: belowTheLineColor },
+      ...belowTheLine,
+    ];
+
+    // We render the default, but with our overridden payload.
+    return <DefaultTooltipContent {...props} payload={newPayload} />;
+  };
+
+  return (
+    <div className="library-stats">
+      {library ? (
+        <h2>{libraryName || libraryKey} Statistics</h2>
+      ) : (
+        <h2>Statistics for All Libraries</h2>
+      )}
+      <ul className="stats">
+        <li className="stat-group">{patronsGroup(patrons)}</li>
+        <li className="stat-group">{circulationsGroup(patrons)}</li>
+        <li className="stat-group">{inventoryGroup(inventory)}</li>
+        <li className="stat-group stat-group-wide">
+          {collectionsGroup(chartItems, CustomTooltip)}
+        </li>
```

Comment:
I factored these out into functions because nesting them inline made it harder to understand what is happening. I wasn't sure whether it would be better to turn them into components, but it seemed easier to call the functions and not have to deal with the props. They aren't used anywhere else.